VMEncryption/main/check_util.py

```diff
@@ -185,11 +185,22 @@ def validate_lvm_os(self, public_settings):
         if detected:
             raise Exception("LVM OS disk layout does not satisfy prerequisites ( see https://aka.ms/adelvm )")
 
-    def precheck_for_fatal_failures(self, public_settings):
+    def validate_memory_os_encryption(self, public_settings, encryption_status):
+        is_enable_operation = False
+        encryption_operation = public_settings.get(CommonVariables.EncryptionEncryptionOperationKey)
+        if encryption_operation in [CommonVariables.EnableEncryption, CommonVariables.EnableEncryptionFormat, CommonVariables.EnableEncryptionFormatAll]:
+            is_enable_operation = True
+        volume_type = public_settings.get(CommonVariables.VolumeTypeKey)
+        if is_enable_operation and not volume_type.lower() == CommonVariables.VolumeTypeData.lower() and encryption_status["os"] == "NotEncrypted":
+            if self.is_insufficient_memory():
+                raise Exception("Not enough memory for enabling encryption on OS volume. At least 7 GB memory is required.")
```

Comment:
7GB is a bare minimum requirement, 8GB is recommended.  Also the available free memory is important, a system may have total RAM but due to other processes not have sufficient free space.   Maybe change "At least 7GB memory is required" to "8 GB memory is recommended."